src/Journal.cpp

```diff
@@ -0,0 +1,331 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  Journal.cpp
+
+  Paul Licameli
+
+*******************************************************************//*!
+
+\namespace Journal
+\brief Facilities for recording and playback of sequences of user interaction
+
+*//*******************************************************************/
+
+#include "Journal.h"
+
+#include <algorithm>
+#include <unordered_map>
+#include <wx/app.h>
+#include <wx/filename.h>
+#include <wx/textfile.h>
+
+#include "MemoryX.h"
+#include "Prefs.h"
+
+namespace {
+
+constexpr auto SeparatorCharacter = ',';
+constexpr auto CommentCharacter = '#';
+
+wxString sFileNameIn;
+wxTextFile sFileIn;
+
+wxString sLine;
+// Invariant:  the input file has not been opened, or else sLineNumber counts
+// the number of lines consumed by the tokenizer
+int sLineNumber = -1;
+
+struct FlushingTextFile : wxTextFile {
+   // Flush output when the program quits, even if that makes an incomplete
+   // journal file without an exit
+   ~FlushingTextFile() { if ( IsOpened() ) { Write(); Close(); } }
```

Comment:
I expected that wxFile will do it, but well. Also, wxTextFile::Write overload frightens me :)

Still, all the fuss here goes to wxWidgets, not to this code